qbraid_qir/squin/example.ipynb

```diff
@@ -0,0 +1,378 @@
+      "\u001b[90m  │ \u001b[0m      \u001b[34msquin.gate\u001b[0m.cx(\u001b[33mcontrols=\u001b[0m%19, \u001b[33mtargets=\u001b[0m%22)\n",
+      "\u001b[90m  │ \u001b[0m%23 = \u001b[34mfunc\u001b[0m.const.none()\u001b[30m : \u001b[0m\u001b[30m!py.NoneType\u001b[0m\n",
+      "\u001b[90m  │ \u001b[0m      \u001b[34mfunc\u001b[0m\u001b[31m.\u001b[0m\u001b[31mreturn\u001b[0m %23\n",
+      "}\u001b[90m // func.func main\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Convert Cirq circuit to squin kernel using bloqade\n",
+    "main_loaded = lowering_module.load_circuit(circuit, kernel_name=\"main\")\n",
+    "\n",
+    "print(\"\\nSquin Kernel (from bloqade/cirq):\")\n",
+    "print(\"=\" * 60)\n",
+    "main_loaded.print()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1cbc7ec3",
+   "metadata": {},
+   "source": [
+    "## Run on Simulator and Compare output vectors"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "0b8735b3",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "cirq-squin-output:\n",
+      "[(0.6854531168937683-0.17364905774593353j), (-0+0j), -0j, -0j, 0j, 0j, 0j, (-0+0j), (-0+0j), 0j, 0j, 0j, 0j, 0j, 0j, (0.6854531168937683-0.17364905774593353j)]\n",
+      "pyqir-squin-output:\n",
+      "[(0.07478517293930054-0.7031408548355103j), 0j, -0j, -0j, 0j, 0j, 0j, 0j, 0j, (-0+0j), 0j, 0j, 0j, 0j, 0j, (0.07478517293930054-0.7031408548355103j)]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"cirq-squin-output:\")\n",
+    "cirq_sim = StackMemorySimulator(min_qubits=4)\n",
+    "cirq_result = cirq_sim.run(main_loaded)\n",
+    "print(cirq_sim.state_vector(main_loaded))\n",
+    "\n",
+    "\n",
+    "print(\"pyqir-squin-output:\")\n",
+    "pyq_sim = StackMemorySimulator(min_qubits=4)\n",
+    "pyq_result = pyq_sim.run(squin_mod)\n",
+    "print(pyq_sim.state_vector(squin_mod))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b5013851",
+   "metadata": {},
+   "source": [
+    "## Actual GHZ Squin Code"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "17cb0f56",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from bloqade import squin\n",
+    "from bloqade.pyqrack import StackMemorySimulator\n",
+    "\n",
+    "\n",
+    "@squin.kernel\n",
+    "def ghz(n: int):\n",
+    "    q = squin.qalloc(n)\n",
+    "    squin.h(q[0])\n",
+    "    for i in range(1, n):\n",
+    "        squin.cx(q[i - 1], q[i])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "73672091",
+   "metadata": {},
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "9c6004b6",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
```

Comment:
Ideally, all the three kernels generated by cirq-> squin, pyqir -> squin AND native squin should produce exact same state vectors , if we are going in for ideal noiseless simulations. Can you confirm why that is? 

Is pyqrack `StackMemorySimulator` simulating noise as well? If yes, then we need to confirm that the majority counts for measurement outcomes lie under the same buckets across some random number of tests. Eg - 

4 qubit GHZ - 0000 and 1111 are the two most probable outcomes